File: Client/src/app/shared/models/Basket.ts
import { v4 as uuidv4 } from 'uuid';
import { BasketItem } from './BasketItem';

export class Basket {
  id: string = uuidv4();
  subTotal: number = 0;
  shippingPrice: number = 0;
  total: number = 0;
  items: BasketItem[] = [];
  deliveryMethodId?: string;
  clientSecret?: string;
  paymentIntentId?: string;
}

File: Domain/Entities/Basket.cs
namespace Domain.Entities;

public class Basket
{
  public Basket() { }

  public Basket(string id)
  {
    Id = id;
  }

  public string Id { get; set; }
  public decimal SubTotal { get; set; }
  public decimal ShippingPrice { get; set; }
  public decimal Total { get; set; }
  public List<BasketItem> Items { get; set; } = new();
  public Guid? DeliveryMethodId { get; set; }
  public string ClientSecret { get; set; }
  public string PaymentIntentId { get; set; }
}
